l10n_es_aeat/models/l10n_es_aeat_report_tax_mapping.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# Copyright 2016-2017 Tecnativa - Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, exceptions, fields, models
+
+
+class L10nEsAeatReportTaxMapping(models.AbstractModel):
+    _name = "l10n.es.aeat.report.tax.mapping"
+    _inherit = "l10n.es.aeat.report"
+    _description = (
+        "Inheritable abstract model to add taxes by code mapping in any AEAT report"
+    )
+
+    tax_line_ids = fields.One2many(
+        comodel_name="l10n.es.aeat.tax.line",
+        inverse_name="res_id",
+        domain=lambda self: [("model", "=", self._name)],
+        auto_join=True,
+        readonly=True,
+        string="Tax lines",
+    )
+
+    def calculate(self):
+        res = super().calculate()
+        for report in self:
+            report.tax_line_ids.unlink()
+            # Buscar configuración de mapeo de impuestos
+            tax_code_map = self.env["l10n.es.aeat.map.tax"].search(
+                [
+                    ("model", "=", report.number),
+                    "|",
+                    ("date_from", "<=", report.date_start),
+                    ("date_from", "=", False),
+                    "|",
+                    ("date_to", ">=", report.date_end),
+                    ("date_to", "=", False),
+                ],
+                limit=1,
+            )
+            if tax_code_map:
+                tax_lines = []
+                for map_line in tax_code_map.map_line_ids:
+                    tax_lines.append(report._prepare_tax_line_vals(map_line))
+                report.tax_line_ids = [(0, 0, x) for x in tax_lines]
+        return res
+
+    def unlink(self):
+        self.mapped("tax_line_ids").unlink()
+        return super().unlink()
+
+    def _prepare_tax_line_vals(self, map_line):
+        self.ensure_one()
+        move_lines = self._get_tax_lines(self.date_start, self.date_end, map_line)
+        if map_line.sum_type == "credit":
+            amount = sum(move_lines.mapped("credit"))
+        elif map_line.sum_type == "debit":
+            amount = sum(move_lines.mapped("debit"))
+        else:  # map_line.sum_type == 'both'
+            amount = sum(move_lines.mapped("credit")) - sum(move_lines.mapped("debit"))
+        if map_line.inverse:
+            amount = (-1.0) * amount
+        return {
+            "model": self._name,
+            "res_id": self.id,
+            "map_line_id": map_line.id,
+            "amount": amount,
+            "move_line_ids": [(6, 0, move_lines.ids)],
+        }
+
+    def _get_partner_domain(self):
+        return []
+
```

Comment:
Se añade la función para la creación del domain de cuentas

`


    def _account_codes_to_domain(self, account_codes):
        """Convert a comma separated list of account codes
        (possibly with wildcards) to a domain on account.account.
         """
        elems = []
        for account_code in account_codes.split(","):
            account_code = account_code.strip()
                if "%" in account_code:
                    elems.append([("account_id.code", "=like", account_code)])
                else:
                    elems.append([("account_id.code", "=", account_code)])
        return tuple(expression.OR(elems))
`